packages/docmanager/src/widgetmanager.ts

```diff
@@ -196,7 +209,9 @@ export class DocumentWidgetManager implements IDisposable {
     if (!factory) {
       return undefined;
     }
-    let newWidget = this.createWidget(factory, context);
+    let newWidget = factory.clone(context);
+    Private.factoryProperty.set(newWidget, factory);
```

Comment:
This `factoryProperty` looks like it could also be set in the `_initializeWidget` function.

Reply:
Thanks, done